packages/devtools_server/lib/src/server.dart

```diff
@@ -77,13 +85,15 @@ Future<HttpServer> serveDevToolsWithArgs(List<String> arguments,
   final bool launchBrowser = args[argLaunchBrowser];
   final bool enableNotifications = args[argEnableNotifications];
   final port = args[argPort] != null ? int.tryParse(args[argPort]) ?? 0 : 0;
+  final bool headlessMode = args[argHeadlessMode];
```

Comment:
Do you need to set the default value here? E.g. ?? false

I think if you pass in an explicit null for headlessMode in the call to serveDevTools, the default in the constructor won't be set.

Reply:
I think the ArgParser is taking care of this and populating `false`s into the things we register as `flags`. It seems to work, and there are many others above (launch browser, enable notifications, machine mode) the same.

(if you think I might be mistaken, lmk though and I can check again - wouldn't be the first time! :-)

Reply:
okay that sounds good! 👍

Reply:
It turned out there was actually a very similar bug in this PR! :( I was passing an explicit `null` as `args` to `Chrome.start` and it assumes it's always a list (since it has a default value of an empty list that this overrides).

I've opened a PR here to fix -> https://github.com/flutter/devtools/pull/1139